pkg/controller/perconaservermongodb/psmdb_controller.go

```diff
@@ -467,6 +473,44 @@ func (r *ReconcilePerconaServerMongoDB) Reconcile(request reconcile.Request) (re
 	return rr, nil
 }
 
+func (r *ReconcilePerconaServerMongoDB) checkFinalizers(cr *api.PerconaServerMongoDB) error {
+	var err error = nil
+	if cr.ObjectMeta.DeletionTimestamp != nil {
+		finalizers := []string{}
+		for _, f := range cr.GetFinalizers() {
+			switch f {
+			case "delete-psmdb-pvc":
+				err = r.deleteAllPVC(cr)
```

Comment:
should we delete StatefulSet first, before PVC?
is racing condition possible? like StatefulSet recreate Pods/PVCs

Reply:
Should we delete StatefulSets explicitly? Because I can't find a place where it's done.

Reply:
in pxc we delete statefulset instead of pvc
https://github.com/percona/percona-xtradb-cluster-operator/blob/master/pkg/controller/pxc/controller.go#L269

Reply:
done, now StatefulSets will be deleted before PVCs.